lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java

```diff
@@ -0,0 +1,158 @@
+    metadataPrefix = config.hasPath("metadata_prefix") ? config.getString("metadata_prefix") : "tika";
+    tikaConfigPath = config.hasPath("tika_config_path") ? config.getString("tika_config_path") : null;
+    textContentLimit = config.hasPath("text_content_limit") ? config.getInt("text_content_limit") : Integer.MAX_VALUE;
+    metadataWhitelist = config.hasPath("metadata_whitelist") ? config.getStringList("metadata_whitelist") : null;
+    metadataBlacklist = config.hasPath("metadata_blacklist") ? config.getStringList("metadata_blacklist") : null;
+    if (metadataWhitelist != null && metadataBlacklist != null) {
+      throw new StageException("Provided both a whitelist and blacklist to the TextExtractor stage");
+    }
+    parseCtx = new ParseContext();
+  }
+
+  @Override
+  public void start() throws StageException {
+    if (this.tikaConfigPath == null) {
+      parser = new AutoDetectParser();
+    } else {
+      try {
+        File f = new File(this.tikaConfigPath);
+        TikaConfig tc = new TikaConfig(f);
+        parser = new AutoDetectParser(tc);
+      } catch (Exception e) {
+        throw new StageException("Error starting TextExtractor stage.", e);
+      }
+    }
+    parseCtx.set(Parser.class, parser);
+  }
+
+  @Override
+  public Iterator<Document> processDocument(Document doc) throws StageException {
+    // if the document has both a byteArray field and a filePathField, only byteArray will be processed.
+    if (doc.has(byteArrayField)) {
+
+      byte[] byteArray = doc.getBytes(byteArrayField);
+
+      InputStream inputStream = new ByteArrayInputStream(byteArray);
+      parseInputStream(doc, inputStream);
+
+    } else if (doc.has(filePathField)) {
+
+      String filePath = doc.getString(filePathField);
+
+      try {
+        InputStream inputStream = FileUtils.getInputStream(filePath);
+        parseInputStream(doc, inputStream);
+      } catch (IOException e) {
+        throw new StageException("InputStream cannot be parsed or created", e);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Cleans the name of metadata field names to be in line with general standards for documents
+   */
+  private static String cleanFieldName(String name) {
+    String cleanName = name.trim().toLowerCase();
+    cleanName = cleanName.replaceAll(" ", "_");
+    cleanName = cleanName.replaceAll("-", "_");
+    cleanName = cleanName.replaceAll(":", "_");
+    return cleanName;
+  }
+
+  /**
+   * Parses given input stream and adds the text data and metadata to given document
+   */
+  public void parseInputStream(Document doc, InputStream inputStream) {
+    Metadata metadata = new Metadata();
+    ContentHandler bch = new BodyContentHandler(textContentLimit);
+    try {
+      parser.parse(inputStream, bch, metadata, parseCtx);
+    } catch (IOException | SAXException | TikaException e) {
+      log.warn("Tika Exception: {}", e.getMessage());
+    }
+
+    doc.addToField(textField, bch.toString());
+    parseCtx.toString();
```

Comment:
what is this doing?

Reply:
Seems to be left behind from debugging, removed in new commit